oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/ttl/RecordTTLCalculator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.storage.ttl;
+
+import org.apache.skywalking.oap.server.core.DataTTLConfig;
+import org.joda.time.DateTime;
+
+/**
+ * Calculate TTL for record.
+ *
+ * @author wusheng
+ */
+public class RecordTTLCalculator implements TTLCalculator {
+
+    @Override public long timeBefore(DateTime currentTime, DataTTLConfig dataTTLConfig) {
+        return Long.valueOf(currentTime.plusMinutes(0 - dataTTLConfig.getRecordDataTTL()).toString("yyyyMMddHHmm"));
```

Comment:
Segment time bucket is the second time format.

Reply:
Fixed, w/ wrong in EsRecordTTLCalculator too, we should use `#plusDays` there, right?

Reply:
Alarm record time bucket is also the second time format.

Alarm
```
record.setStartTime(message.getStartTime());
record.setTimeBucket(TimeBucket.getSecondTimeBucket(message.getStartTime()));
``` 

Segment
```
long timeBucket = TimeBucket.getSecondTimeBucket(segmentCoreInfo.getStartTime());
segment.setTimeBucket(timeBucket);
```

Suggest that, rename the getSecondTimeBucket to getRecordTimeBucket or create an alias name of getRecordTimeBucket, it will make it clear.

Reply:
> Fixed, w/ wrong in EsRecordTTLCalculator too, we should use `#plusDays` there, right?

En, En.

Reply:
toString("yyyyMMddHHmm") is wrong, modify to  toString("yyyyMMddHHmmss").

Reply:
Plus minute is for the config item in application.yml. 
 `recordDataTTL: ${SW_CORE_RECORD_DATA_TTL:90} # Unit is minute`

toString("yyyyMMddHHmmss") if for the database, the `timeBucket` column in database is second data format.